From 0f8f894bb743feebd6b21348380f4f176d3a474c Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Thu, 12 Nov 2020 17:50:54 +0800
Subject: [PATCH] phy: phy-rockchip-naneng-usb2: update USB_DCP state if cable
 plug-out

The *USB_DCP* should be updated when the cable plug out from dedicated
charging port. As a result, the charger could set the charging current
by *EXTCON_CHG_USB_DCP*.

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
Change-Id: I478e584552a627de928378c6da1e6392a8fb04c4
---
 drivers/phy/rockchip/phy-rockchip-naneng-usb2.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c b/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
index f3a89f80a458..a91e60293a92 100644
--- a/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
@@ -951,8 +951,11 @@ static void rockchip_usb2phy_otg_sm_work(struct work_struct *work)
 			break;
 		}
 	} else {
-		if (!rport->perip_connected)
+		if (!rport->perip_connected) {
+			if (extcon_get_state(rphy->edev, EXTCON_CHG_USB_DCP) > 0)
+				extcon_set_state_sync(rphy->edev, EXTCON_CHG_USB_DCP, 0);
 			return;
+		}
 
 		dev_dbg(&rport->phy->dev, "usb peripheral disconnect\n");
 		wake_unlock(&rport->wakelock);
-- 
2.35.3

